src/worker/mod.rs

```diff
@@ -20,7 +20,12 @@ pub(crate) const STATUS_RUNNING: usize = 0;
 pub(crate) const STATUS_QUIET: usize = 1;
 pub(crate) const STATUS_TERMINATING: usize = 2;
 
-type CallbacksRegistry<E> = FnvHashMap<String, runner::BoxedJobRunner<E>>;
+pub(crate) enum Callback<E> {
+    Async(runner::BoxedJobRunner<E>),
+    Sync(Arc<dyn Fn(Job) -> Result<(), E> + Sync + Send + 'static>),
```

Comment:
I think this can be `Box<dyn` since the callbacks are already stored in an `Arc`

Reply:
Please see [comment](https://github.com/jonhoo/faktory-rs/pull/65#discussion_r1615257651)